extensions/spark/install_spark.sh

```diff
@@ -128,7 +128,6 @@ export SPARK_DAEMON_MEMORY=${SPARK_DAEMON_MEMORY}m
 export SPARK_WORKER_DIR=${SPARK_WORKDIR}
 export SPARK_LOCAL_DIRS=${SPARK_TMPDIR}
 export SPARK_LOG_DIR=${SPARK_LOG_DIR}
-export SPARK_CLASSPATH=\$SPARK_CLASSPATH:${LOCAL_GCS_JAR}
```

Comment:
Right, there are a couple bugs with this. Just removing SPARK_CLASSPATH for the GCS connector without adding ${LOCAL_GCS_JAR} to both spark.executor.extraClassPath and spark.driver.extraClassPath causes the GCS connector to not work; you can test GCS-connector functionality with a bdutil-deployed clusters by running `./bdutil <your bdutil flags> shell < extensions/spark/spark-validate-setup.sh`. With these changes, even without including bigtable_env.sh, the following error:

    15/07/06 18:41:42 INFO BlockManagerMasterActor: Registering block manager dhuo-spark2-m.c.hadoop-cloud-dev.google.com.internal:53234 with 1949.9 MB RAM, BlockManagerId(<driver>, dhuo-spark2-m.c.hadoop-cloud-dev.google.com.internal, 53234)
    15/07/06 18:41:42 INFO BlockManagerMaster: Registered BlockManager
    java.lang.RuntimeException: java.lang.ClassNotFoundException: com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystem
      at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:867)
      at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:1380)
      at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:66)
      at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:1404)
      at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:254)
      at org.apache.spark.util.Utils$.getHadoopFileSystem(Utils.scala:1522)
      at org.apache.spark.scheduler.EventLoggingListener.<init>(EventLoggingListener.scala:64)
      at org.apache.spark.SparkContext.<init>(SparkContext.scala:400)
      at org.apache.spark.repl.SparkILoop.createSparkContext(SparkILoop.scala:1016)
      at $iwC$$iwC.<init>(<console>:9)
      at $iwC.<init>(<console>:18)
      at <init>(<console>:20)
      at .<init>(<console>:24)
      at .<clinit>(<console>)
      at .<init>(<console>:7)
      at .<clinit>(<console>)

The other issue is that even if you add the following to `spark-defaults.conf`:

    spark.executor.extraClassPath ${LOCAL_GCS_JAR}
    spark.driver.extraClassPath ${LOCAL_GCS_JAR}

This will make spark-validate-setup.sh succeed, but the EventLog is then broken with:

    5/07/06 18:54:22 ERROR Master: Exception in replaying log for application Spark shell!
    java.lang.RuntimeException: java.lang.ClassNotFoundException: com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystem
      at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:867)
      at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:1380)
      at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:66)
      at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:1404)
      at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:254)
      at org.apache.spark.util.Utils$.getHadoopFileSystem(Utils.scala:1522)
      at org.apache.spark.util.Utils$.getHadoopFileSystem(Utils.scala:1529)
      at org.apache.spark.deploy.master.Master.rebuildSparkUI(Master.scala:747)
      at org.apache.spark.deploy.master.Master.removeApplication(Master.scala:710)
      at org.apache.spark.deploy.master.Master.finishApplication(Master.scala:688)
      at org.apache.spark.deploy.master.Master$$anonfun$receiveWithLogging$1$$anonfun$applyOrElse$29.apply(Master.scala:432)
      at org.apache.spark.deploy.master.Master$$anonfun$receiveWithLogging$1$$anonfun$applyOrElse$29.apply(Master.scala:432)
      at scala.Option.foreach(Option.scala:236)
      at org.apache.spark.deploy.master.Master$$anonfun$receiveWithLogging$1.applyOrElse(Master.scala:432)
      at scala.runtime.AbstractPartialFunction$mcVL$sp.apply$mcVL$sp(AbstractPartialFunction.scala:33)
      at scala.runtime.AbstractPartialFunction$mcVL$sp.apply(AbstractPartialFunction.scala:33)
      at scala.runtime.AbstractPartialFunction$mcVL$sp.apply(AbstractPartialFunction.scala:25)
      at org.apache.spark.util.ActorLogReceive$$anon$1.apply(ActorLogReceive.scala:53)
      at org.apache.spark.util.ActorLogReceive$$anon$1.apply(ActorLogReceive.scala:42)
      at scala.PartialFunction$class.applyOrElse(PartialFunction.scala:118)
      at org.apache.spark.util.ActorLogReceive$$anon$1.applyOrElse(ActorLogReceive.scala:42)
      at akka.actor.Actor$class.aroundReceive(Actor.scala:465)
      at org.apache.spark.deploy.master.Master.aroundReceive(Master.scala:52)
      at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516)
      at akka.actor.ActorCell.invoke(ActorCell.scala:487)
      at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:238)
      at akka.dispatch.Mailbox.run(Mailbox.scala:220)
      at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393)
      at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
      at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
      at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
      at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
    Caused by: java.lang.ClassNotFoundException: com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystem
      at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
      at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
      at java.security.AccessController.doPrivileged(Native Method)
      at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
      at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
      at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
      at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
      at java.lang.Class.forName0(Native Method)
      at java.lang.Class.forName(Class.java:274)
      at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:820)
      at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:865)
      ... 31 more

Running `ps wwaux` shows in this case the Master indeed not picking up the classpaths:

    hadoop   24535  1.9  1.3 2889160 208976 ?      Sl   18:51   0:07 java -cp /home/hadoop/spark-install/sbin/../conf:/home/hadoop/spark-install/lib/spark-assembly-1.3.1-hadoop1.0.4.jar:/home/hadoop/spark-install/lib/datanucleus-core-3.2.10.jar:/home/hadoop/spark-install/lib/datanucleus-rdbms-3.2.9.jar:/home/hadoop/spark-install/lib/datanucleus-api-jdo-3.2.6.jar -XX:MaxPermSize=128m -Dspark.akka.logLifecycleEvents=true -Xms2261m -Xmx2261m org.apache.spark.deploy.master.Master --ip dhuo-spark3-m --port 7077 --webui-port 8080

There's probably a way to work around this issue. In any case, the GCS connector must be present on the classpath of all three of: Daemons, Driver, and Executors, or the cluster will not function properly.